test/blackbox/common/RTPSBlackboxTestsBasic.cpp

```diff
@@ -568,6 +569,64 @@ TEST_P(RTPS, RTPSAsReliableTransientLocalTwoWritersConsecutives)
     writer.destroy();
 }
 
+/**
+ * This test checks the addition of network interfaces at run-time.
+ *
+ * After launching the reader with the network interfaces enabled,
+ * the writer is launched with the transport simulating that there
+ * are no interfaces.
+ * No participant discovery occurs.
+ *
+ * In a second step, the flag to simulate no interfaces is disabled and
+ * RTPSParticipant::update_attributes() called to add the "new" interfaces.
+ * Discovery is succesful and communication is established.
+ */
+TEST(RTPS, RTPSNetworkInterfaceChangesAtRunTime)
+{
+    RTPSWithRegistrationReader<HelloWorldType> reader(TEST_TOPIC_NAME);
+    RTPSWithRegistrationWriter<HelloWorldType> writer(TEST_TOPIC_NAME);
+
+    // reader is initialized with all the network interfaces
+    reader.init();
+    ASSERT_TRUE(reader.isInitialized());
+
+    // writer: launch without interfaces
+    test_UDPv4Transport::simulate_no_interfaces = true;
+    auto test_transport = std::make_shared<test_UDPv4TransportDescriptor>();
+    writer.disable_builtin_transport().add_user_transport_to_pparams(test_transport).init();
+    ASSERT_TRUE(writer.isInitialized());
+
+    // no discovery
+    writer.wait_discovery(std::chrono::seconds(3));
+    reader.wait_discovery(std::chrono::seconds(3));
+
+    EXPECT_FALSE(writer.is_matched());
+    EXPECT_FALSE(reader.is_matched());
+
```

Comment:
I'm thinking on whether we should send here and wait in the reader for a while, then reload the interfaces and wait on the reader again. First time nothing will be received, second everything. What do you think?

Reply:
If there is no matching there is not going to be any communication. So if the reader is `VOLATILE` as it is now (default) even though the interface is enabled later and discovery succeeds, no data is going to be received in the reader.

Reply:
Data is going to be sent and the reader is going to be `TRANSIENT_LOCAL` to access the already sent data